### ISSUE TITLE
Change detected when defining "azuredevops_pipeline_authorization" resources on multiple build_definitions

### ISSUE THREAD

**Issue by HammerNL89** [microsoft/terraform-provider-azuredevops#821]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version
Terraform v1.5.2
ADO provider = 0.6.0

### Affected Resource(s)

* `azuredevops_pipeline_authorization`

### Terraform Configuration Files

In short, I define two buid_definitions and one service connection, I would like to authorize the sc to be used on both definitions.

```hcl
...
resource "azuredevops_build_definition" "deploy" {
  project_id = data.azuredevops_project.main.id
  name       = "DEPLOY"
  path       = "\\DEPLOY"

  repository {
    repo_type   = "TfsGit"
    repo_id     = data.azuredevops_git_repository.main.id
    branch_name = "master"
    yml_path    = "cicd/deploy.yaml"
  }
}

resource "azuredevops_pipeline_authorization" "deploy" {
  project_id  = data.azuredevops_project.main.id
  resource_id = azuredevops_serviceendpoint_azurerm.main.id
  type        = "endpoint"
  pipeline_id = azuredevops_build_definition.deploy.id
}

resource "azuredevops_build_definition" "destroy" {
  project_id = data.azuredevops_project.main.id
  name       = "DESTROY"
  path       = "\\DESTROY"

  repository {
    repo_type   = "TfsGit"
    repo_id     = data.azuredevops_git_repository.main.id
    branch_name = "master"
    yml_path    = "cicd/destroy.yaml"
  }
}

resource "azuredevops_pipeline_authorization" "destroy" {
  project_id  = data.azuredevops_project.main.id
  resource_id = azuredevops_serviceendpoint_azurerm.main.id
  type        = "endpoint"
  pipeline_id = azuredevops_build_definition.destroy.id
}

```

###  Output
2nd run
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # azuredevops_pipeline_authorization.destroy must be replaced
-/+ resource "azuredevops_pipeline_authorization" "destroy" {
      ~ id          = "90877aaf-a793-470a-a3c8-04f36019c231" -> (known after apply)
      ~ pipeline_id = 10077 -> 10076 # forces replacement
        # (3 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```
3rd run
```
Terraform will perform the following actions:

  # azuredevops_pipeline_authorization.deploy must be replaced
-/+ resource "azuredevops_pipeline_authorization" "deploy" {
      ~ id          = "90877aaf-a793-470a-a3c8-04f36019c231" -> (known after apply)
      ~ pipeline_id = 10076 -> 10077 # forces replacement
        # (3 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

### Expected Behavior

2nd run of terraform apply should not introduce changes
### Actual Behavior

First run applies okay, every run afterwards introduces a changes / flip between the two `azuredevops_pipeline_authorization` resources. 

### Steps to Reproduce

1. `terraform apply`

### Important Factoids
The deprecated resource `azuredevops_resource_authorization` doesn't have this issue. 

### References

* #0000

**Comment by xuzhang3** [comment]
@HammerNL89  This is bug, fixed by #809

**Comment by HammerNL89** [comment]
> @HammerNL89 This is bug, fixed by #809

Okay, I thought that was different, because I specify a `pipeline_id`, but my scenario is also covered?

**Comment by xuzhang3** [comment]
@HammerNL89 yes, former implementation will always get the first authorized pipeline as the `pipeline_id`.
